doc/releases/migration-guide-4.1.rst

```diff
@@ -131,6 +131,11 @@ Stepper
 Regulator
 =========
 
+UART
```

Comment:
This section has been named `SERIAL` in past release notes- otherwise, change LGTM